sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -0,0 +1,1422 @@
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public TextAnalyticsClient(Uri endpoint, TokenCredential credential, TextAnalyticsClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            Argument.AssertNotNull(options, nameof(options));
+
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="subscriptionKey"></param>
+        public TextAnalyticsClient(Uri endpoint, string subscriptionKey)
+            : this(endpoint, subscriptionKey, new TextAnalyticsClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="subscriptionKey"></param>
+        /// <param name="options"></param>
+        public TextAnalyticsClient(Uri endpoint, string subscriptionKey, TextAnalyticsClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNullOrEmpty(subscriptionKey, nameof(subscriptionKey));
+            Argument.AssertNotNull(options, nameof(options));
+
+            _baseUri = endpoint;
+            _subscriptionKey = subscriptionKey;
+            _apiVersion = options.GetVersionString();
+            _pipeline = HttpPipelineBuilder.Build(options);
+            _clientDiagnostics = new ClientDiagnostics(options);
+        }
+
+        #region Detect Language
+
+        // Note that this is a simple overload that takes a single input and returns a single detected language.
+        // More advanced overloads are available that return the full list of detected languages.
+        /// <summary>
+        /// </summary>
+        /// <param name="inputText"></param>
+        /// <param name="countryHint"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<DetectedLanguage>> DetectLanguageAsync(string inputText, string countryHint = "us", CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(inputText, nameof(inputText));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.AI.TextAnalytics.TextAnalyticsClient.DetectLanguage");
+            scope.AddAttribute("inputText", inputText);
+            scope.Start();
+
+            try
+            {
+                using Request request = CreateDetectLanguageRequest(new List<string> { inputText }, countryHint);
+                Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+                switch (response.Status)
```

Comment:
Just a tip since this gets repeated a lot: maybe a generic helper function that handles common return codes, throws errors, and returns factoried type (we do something like that in KV). Helps eliminate dups.

Reply:
Yes, I was admiring what KV has done around this, and excited to refactor to similar patterns.  #8711